Completed the docs + generate-keys improvement

ezimuel opened the pull request:
This PR adds the OAuth2 documentation and improves the `bin/generate-keys.php` script.

ezimuel committed:
Completed the docs + generate-keys improvement

ezimuel commented:
Ping @weierophinney for review, thanks!

mwillbanks commented on docs/book/grant/auth_code.md:
```diff
@@ -0,0 +1,69 @@
+
+## Request the access token
+
+The client send a POST request to the authorization server with the following
```
Change `send` to `sends`

mwillbanks commented on docs/book/grant/client_credentials.md:
```diff
@@ -0,0 +1,27 @@
+# Client credentials
+
+The client credentials grant is used in machine-to-machine scenarios, for
+example a client making API requests that do not require user’s permission.
```
`not require user's` to `not require a user's`

mwillbanks commented on docs/book/grant/client_credentials.md:
```diff
@@ -0,0 +1,27 @@
+
+The `token_type` is the type of generated token (Bearer). The `expires_in` is
+an integer representing the TTL (in seconds) of the access token.
+The `access_token` contains the a `JWT` signed with the authorization server’s
```
`contains the a` to `contains a`.

mwillbanks commented on docs/book/grant/implicit.md:
```diff
@@ -0,0 +1,33 @@
+- `response_type` = token;
+- `client_id` with the client’s ID;
+- `redirect_uri` with the client redirect URI. This parameter is optional,
+  but if not send the user will be redirected to a pre-registered redirect URI;
```
`but if not send the user will be` to `if not, the user will be`

mwillbanks commented on docs/book/grant/refresh_token.md:
```diff
@@ -0,0 +1,34 @@
+an integer representing the TTL (in seconds) of the access token.
+The `refresh_token` a token that can be used to refresh the `access_token` when
+expired.
+The `access_token` contains the a `JWT` signed with the authorization server’s
```
`contains the a` to `contains a`

mwillbanks reviewed (changes requested):
Looks good to me, just some cleanup to do on the documentation wording.

ezimuel committed:
Fixed typos

ezimuel commented:
@mwillbanks I just updated the PR with your feedbacks. Thanks for reviewing!

ezimuel committed:
Fixed typo on implicit doc

mbourquin commented:
@ezimuel May be add a little note about the way how to generate a password or client secret? I know it is already in Apigility documentation but not directly in this doc. (bin/bcrypt.php in Apigility) or add a new script in bin/ ?

weierophinney committed:
Merge pull request #19 from ezimuel/feature/completed-docs

Completed the docs + generate-keys improvement

weierophinney committed:
Edits for #19

Adds a number of edits, clarifying acronyms, updating wording, and
fixing grammar.

weierophinney commented:
@ezimuel I've pushed a few changes, but otherwise, :+1: !
